stac_pydantic/item.py

```diff
@@ -28,10 +28,15 @@ def validate_datetime(cls, v, values):
                 raise ValueError(
                     "start_datetime and end_datetime must be specified when datetime is null"
                 )
+
+        if isinstance(v, str):
+            return dt.strptime(v, DATETIME_RFC339)
```

Comment:
because it's a validator maybe it will be good to raise the same error as in https://github.com/arturo-ai/stac-pydantic/pull/62/files#diff-cedfbbe7e0d06171e4f3cdfab0c02e1712111610dc9419080077b4d15a1c3d85R79 🤷‍♂️